### PR TITLE
OCPBUGS-20161: Remove unncessary kas port exposure on service

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
This was first introduced here https://github.com/openshift/hypershift/pull/1614/files#diff-62c16653415b8d89921cb26796abc479c31da1654095f7c46b551b470533d66dR368-R372 not sure with which purpose.

With the changes in https://github.com/openshift/hypershift/pull/2964/files#diff-39a001edaf01802e79e43b99b8ce38585ea7079c7253be01fe33a4902a9ca184 we stopped defaulting the kas container port to 443 resulting in this "if" to evaluate as true and exposing a new port https://github.com/openshift/hypershift/pull/2964/files#diff-62c16653415b8d89921cb26796abc479c31da1654095f7c46b551b470533d66dL277

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
/hold
to make sure of the impact.

**Review by mjlshen — APPROVED** [review]
/lgtm

File: control-plane-operator/controllers/hostedcontrolplane/ingress/router.go
Comment (mjlshen) [inline control-plane-operator/controllers/hostedcontrolplane/ingress/router.go:268]: I think removing this is correct

File: control-plane-operator/controllers/hostedcontrolplane/ingress/router.go
Comment (mjlshen) [inline control-plane-operator/controllers/hostedcontrolplane/ingress/router.go:256]: I'm not familiar with the use-case, but is this so that someone could separate router/KAS traffic onto different ports? If so, this would just be a guard for the ROSA HCP use-case where we use the same port for both. We would then need the `if !foundKas` block further down at line 279

Comment (enxebre) [reply control-plane-operator/controllers/hostedcontrolplane/ingress/router.go:256]: >I'm not familiar with the use-case, but is this so that someone could separate router/KAS traffic onto different ports? If so, this would just be a guard for the ROSA HCP use-case where we use the same port for both. We would then need the if !foundKas block further down at line 279

Might be the original reason, I'm not aware of this use atm.
But also the `if kasPort == 443 {foundKAS = true}` is not a valid criteria any more for that potential use case.

File: control-plane-operator/controllers/hostedcontrolplane/kas/params.go
Comment (sjenning) [inline control-plane-operator/controllers/hostedcontrolplane/kas/params.go:387]: This is probably suppose to be 443 right?

**Review by sjenning — CHANGES_REQUESTED** [review]

**Comment by sjenning** [comment]
waiting to lgtm until after the weekend just in case :sweat_smile:

**Comment by jasonrhunter** [comment]
@sjenning ok to lgtm now?

**Comment by enxebre** [comment]
@mjlshen @sjenning this need to be split in two different PRs and 3 steps:
- HO change https://github.com/openshift/hypershift/pull/3147 PTAL
- Rollout HO
- CPO change

Otherwise environments without this HO, but using the payload with the change might have a data plane haproxy pointing to 6443 which won't be exposed anymore.

**Pull request closed without merging**
